### ISSUE TITLE
Enable Magisk root support.

### ISSUE THREAD

**Issue by ShapeShifter499** [AdAway/AdAway#876]
I've added in Magisk root support but I'm not sure if this is the correct way. https://github.com/AdAway/AdAway/pull/875   

If not added I see the issue described in https://github.com/AdAway/AdAway/issues/845

**Comment by ShapeShifter499** [comment]
I found the cause of my issue, I'll close this one and create a new one.

**Comment by leandro-lucarella-sociomantic** [comment]
Mmm, I wanted to use AdAway with Magisk too and investigating a bit I found the "systemless hosts" option stores a file in `/magisk/.core/hosts`. So I configured AdAway to use that file but it complains that there is no symlink to `/system/etc/hosts` and ask to create one. If I ignore the error (I don't create the link since I don't want to change the `/system` it seems to work fine, but AdAway shows itself as "disabled").

I couldn't find any information about the Magisk "systemless hosts" option (if it mounts the `/system/etc/hosts` from another place so changing it doesn't change the system partition, but it could be nice to have an explicit option in AdAway to deal with this, like one more option in the `hosts` location that is just "Use Magisk" (or even auto-detect if Magisk is installed).
